debezium-server-milvus/src/main/java/io/debezium/server/milvus/MilvusChangeConsumer.java

```diff
@@ -0,0 +1,207 @@
+ *
+ * @author Jiri Pechanec
+ *
+ */
+@Named("milvus")
+@Dependent
+public class MilvusChangeConsumer extends BaseChangeConsumer implements DebeziumEngine.ChangeConsumer<ChangeEvent<Object, Object>> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MilvusChangeConsumer.class);
+
+    private static final String PROP_PREFIX = "debezium.sink.milvus.";
+
+    private MilvusClientV2 milvusClient;
+    private MilvusSchema schema;
+    private final Gson gson = new Gson();
+
+    @ConfigProperty(name = PROP_PREFIX + "uri", defaultValue = "http://localhost:19530")
+    String uri;
+
+    @ConfigProperty(name = PROP_PREFIX + "database", defaultValue = "default")
+    String databaseName;
+
+    @Inject
+    @CustomConsumerBuilder
+    Instance<MilvusClientV2> customClient;
+
+    @PostConstruct
+    void connect() {
+        if (customClient.isResolvable()) {
+            milvusClient = customClient.get();
+            LOGGER.info("Obtained custom configured MilvusClientV2 '{}'", milvusClient);
+        }
+        else {
+            final var config = ConnectConfig.builder()
+                    .uri(uri)
+                    .build();
+            milvusClient = new MilvusClientV2(config);
+            schema = new MilvusSchema(milvusClient);
+        }
+
+        final var databases = milvusClient.listDatabases().getDatabaseNames();
+        if (!databases.contains(databaseName)) {
+            throw new DebeziumException(String.format("Database '%s' does not exist", databaseName));
+        }
+    }
+
+    @PreDestroy
+    void close() {
+        try {
+            milvusClient.close();
+        }
+        catch (Exception e) {
+            LOGGER.warn("Exception while closing client", e);
+        }
+    }
+
+    @Override
+    public void handleBatch(List<ChangeEvent<Object, Object>> records, RecordCommitter<ChangeEvent<Object, Object>> committer)
+            throws InterruptedException {
+
+        for (final ChangeEvent<Object, Object> record : records) {
+            LOGGER.trace("Received event '{}'", record);
+
+            final var sourceRecord = toSourceRecord(record);
+
+            // Milvus does not support dots in collection names so we by default replace them
+            // with underscores so the user does not need to provide router or mapper
+            final var collectionName = streamNameMapper.map(record.destination()).replace('.', '_');
+
+            if (isSchemaChange(sourceRecord)) {
+                LOGGER.debug("Schema change event, ignoring it");
+                committer.markProcessed(record);
+                continue;
+            }
+
+            if (record.key() == null) {
+                throw new DebeziumException("Milvus does not support collections without primary key");
+            }
+            schema.validateKey(collectionName, sourceRecord.keySchema());
+
+            if (sourceRecord.value() == null) {
+                deleteRecord(collectionName, record, committer);
```

Comment:
Do we actually need to handle tombstones? The record should be deleted upon receiving delete event.

Reply:
This is intentional in case `ExtractNewRecordState` SMT is applied. As soon as we agree on handingld of deletes/tombstones in JDBC sink for different siatiations I'll apply the same logic.
Currently I just took the most conservative approach even at the cost of double calls on delete.

Reply:
I see, I forgot about `ExtractNewRecordState`